src/core_modules/capture-core/components/ScopeSelector/ScopeSelector.container.js

```diff
@@ -0,0 +1,195 @@
+// @flow
+import React, { type ComponentType, useEffect, useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useLocation } from 'react-router';
+import { ScopeSelectorComponent } from './ScopeSelector.component';
+import {
+    setOrgUnitFromLockedSelector,
+    setProgramIdFromLockedSelector,
+    setCategoryOptionFromLockedSelector,
+    resetCategoryOptionFromLockedSelector,
+    resetAllCategoryOptionsFromLockedSelector,
+    openNewRegistrationPageFromLockedSelector,
+    openSearchPageFromLockedSelector,
+    fetchOrgUnit,
+    resetProgramIdBatchAction,
+    startAgainBatchAction,
+    resetOrgUnitIdBatchAction,
+} from './ScopeSelector.actions';
+import { resetProgramIdBase } from './QuickSelector/actions/QuickSelector.actions';
+import type { OwnProps } from './ScopeSelector.types';
+import { pageFetchesOrgUnitUsingTheOldWay } from '../../utils/url';
+
+const deriveReadiness = (lockedSelectorLoads, selectedOrgUnitId, organisationUnits) => {
+    // because we want the orgUnit to be fetched and stored
+    // before allowing the user to view the locked selector
+    if (selectedOrgUnitId) {
+        const orgUnit = organisationUnits[selectedOrgUnitId];
+        return Boolean(orgUnit && orgUnit.id && !lockedSelectorLoads);
+    }
+    return !lockedSelectorLoads;
+};
+
+const useUrlQueries = (): { selectedProgramId: string, selectedOrgUnitId: string, pathname: string } =>
+    useSelector(({
+        currentSelections: {
+            programId: selectedProgramId,
+            orgUnitId: selectedOrgUnitId,
+        },
+        router: {
+            location: {
+                query: {
+                    programId: routerProgramId,
+                    orgUnitId: routerOrgUnitId,
+                },
+                pathname,
+            },
+        },
+    }) =>
+        ({
+            selectedProgramId: routerProgramId || selectedProgramId,
+            selectedOrgUnitId: routerOrgUnitId || selectedOrgUnitId,
+            pathname,
```

Comment:
should we add `shallowEqual` here? WDYT?

Reply:
In this PR I just copied old logic inside a new folder/component. The real changes are implemented in https://github.com/dhis2/capture-app/pull/1878. There the `currentSelections` logic is removed. Thanks!